### PR TITLE
Request for new keyboard "Teton 78".

### PR DESCRIPTION
**Pull request by AwesomeBalogh**
## Description

I have designed firmware for my board the Teton 78. Here is my firmware, this is my first time uploading firmware so I apologize for any mistakes. I have read all of the docs so I think this is passable.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x ] I have added tests to cover my changes.
- [ x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mountainmechdesigns/teton_78/readme.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+Make example for this keyboard (after setting up your build environment):
+
+    make teton_78:default
+    make teton_78:via
+    qmk compile -kb teton_78/ -km default
+
+Flashing example for this keyboard:
+
+    make teton_78:default:flash
+    make teton_78:via:flash
```
Comment (waffle87) [inline keyboards/mountainmechdesigns/teton_78/readme.md:20]: A link to the product page would be most preferable here.
```suggestion
* Keyboard Maintainer: [Bennett Balogh (https://github.com/AwesomeBalogh)
* Hardware Supported: Atmega34u4 for Teton 78s' Hotswap and Solder PCBs
* Hardware Availability: [Teton 78 Group Buy](https://www.mountainmechdesigns.com/shop/p/product-5-fb9r2)

Make example for this keyboard (after setting up your build environment):

    make mountainmechdesigns/teton_78:default

Flashing example for this keyboard:

    make mountainmechdesigns/teton_78:default:flash
```

File: keyboards/mountainmechdesigns/teton_78/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+
+    [0] = LAYOUT(
+
+        KC_F1, KC_F6, KC_GRV, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_INS,  KC_HOME, 
+		KC_F2,   KC_F7,    KC_TAB,  KC_Q,    KC_W,   KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL,  KC_END, 
+		KC_F3,   KC_F8,    KC_CAPS, KC_A,    KC_S,   KC_D, KC_F, KC_G, KC_H, KC_J, KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT, 
+		KC_F4,   KC_F9,    KC_LSFT, KC_Z,    KC_X,   KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,                   KC_UP, 
+		KC_F5,   KC_F10,   KC_LCTL, KC_LGUI, KC_LALT,            KC_SPC,                    KC_RALT, KC_RGUI, KC_RCTL, KC_LEFT,          KC_DOWN, KC_RGHT)
```
Comment (waffle87) [inline keyboards/mountainmechdesigns/teton_78/keymaps/default/keymap.c:28]:
```suggestion
        KC_F2,   KC_F7,    KC_TAB,  KC_Q,    KC_W,   KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL,  KC_END, 
        KC_F3,   KC_F8,    KC_CAPS, KC_A,    KC_S,   KC_D, KC_F, KC_G, KC_H, KC_J, KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT, 
        KC_F4,   KC_F9,    KC_LSFT, KC_Z,    KC_X,   KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,                   KC_UP, 
        KC_F5,   KC_F10,   KC_LCTL, KC_LGUI, KC_LALT,            KC_SPC,                    KC_RALT, KC_RGUI, KC_RCTL, KC_LEFT,          KC_DOWN, KC_RGHT)
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/mountainmechdesigns/teton_78/config.h
[the remark this answers is not in the document]
Comment (AwesomeBalogh) [reply keyboards/mountainmechdesigns/teton_78/config.h:1]: Deleted config.h

File: keyboards/mountainmechdesigns/teton_78/keyboard.json
[the remark this answers is not in the document]
Comment (AwesomeBalogh) [reply keyboards/mountainmechdesigns/teton_78/keyboard.json:1]: Done. Thank you for your help!!

File: keyboards/mountainmechdesigns/teton_78/readme.md (hunk 1 above)
Comment (AwesomeBalogh) [reply keyboards/mountainmechdesigns/teton_78/readme.md]: Updated readme :)

File: keyboards/teton_78/info.json
Comment (waffle87) [inline keyboards/teton_78/info.json:1]: Looks like the file and directory made its way in by mistake?

File: keyboards/mountainmechdesigns/teton_78/readme.md
Comment (waffle87) [inline keyboards/mountainmechdesigns/teton_78/readme.md:1]: This readme now appears incomplete, it should closer to how you had it previously. See the [template](https://github.com/qmk/qmk_firmware/blob/master/data/templates/keyboard/readme.md?plain=1) for reference.

File: keyboards/teton_78/info.json
Comment (AwesomeBalogh) [reply keyboards/teton_78/info.json:1]: Whoops sorry, I just deleted that.

File: keyboards/mountainmechdesigns/teton_78/readme.md
Comment (AwesomeBalogh) [reply keyboards/mountainmechdesigns/teton_78/readme.md:1]: Sorry, I just updated that with the missing info. :)

**Review by waffle87 — APPROVED** [review]

File: keyboards/mountainmechdesigns/teton_78/readme.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+The Teton 78 has entered GB and sucessfully finalized in November 2024 The IC page for the keyboard can be found [here](https://geekhack.org/index.php?topic=123275.0).
+The sale page can be found [here](https://www.mountainmechdesigns.com/)
+Discord link can be found [here](https://discord.gg/h9dMwRNfVy)
+
+## About
+
+Keyboard Maintainer: [Bennett Balogh](https://github.com/AwesomeBalogh)
+Hardware Supported: Atmega32u4
+Hardware Availability: N/A
```
Comment (fauxpark) [inline keyboards/mountainmechdesigns/teton_78/readme.md:17]:
```suggestion
* Keyboard Maintainer: [Bennett Balogh](https://github.com/AwesomeBalogh)
* Hardware Supported: Atmega32u4
* Hardware Availability: N/A
```

File: keyboards/mountainmechdesigns/teton_78/readme.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+## How to compile
+
+After setting up your build environment, you can compile the Teton 78 default keymap by using:
+
+    make teton_78:default
+
+Flash using
+
+    make teton_78:default:flash
```
Comment (fauxpark) [inline keyboards/mountainmechdesigns/teton_78/readme.md:27]:
```suggestion
    make mountainmechdesigns/teton_78:default

Flash using

    make mountainmechdesigns/teton_78:default:flash
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mountainmechdesigns/teton_78/readme.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# Teton 78 QMK Firmware
+
+## Introduction
+
+[Teton 78](https://imgur.com/a/alexotos-pictures-of-teton-78-A39gN9e)
```
Comment (zvecr) [inline keyboards/mountainmechdesigns/teton_78/readme.md:5]:
```suggestion
![Teton 78](https://i.imgur.com/MnASIrv.jpeg)
```

**Comment by AwesomeBalogh** [comment]
Just committed the recent suggestion. Thanks!

**Review by fauxpark — APPROVED** [review]

**Pull request merged**